### PR TITLE
Reading GeoTIFFs with rioxarray/rasterio

### PR DESCRIPTION
**Pull request by chpolste**
A basic reader for GeoTIFF files using rioxarray under the hood. To address #120.

- Implements `GeoTIFFReader`, `GeoTIFFFieldList`, `GeoTIFFField`, `GeoTIFFMetadata` and `GeoTIFFGeography`.
- The unit test comes with a small multi-band test file.
- New `geotiff` optional dependencies in `pyproject.toml`.

There is not much in terms of metadata extraction other than the projection information yet, but many GeoTIFFs don't provide much metadata in the first place, unfortunately.

I have configured the reader to apply masking and scaling to the values by default, even though rioxarray doesn't.

Small things adjusted along the way:

- Turned `shape` of the `core.geography.Geography` abstract class into a method (was a property), since that is how it seems to be used in practice.
- Added a `TransverseMercator` projection, which seems to be common for GeoTIFF data.
- Changed `utils.projections.cf.CF_PARAMS_TO_GLOBE_KWARGS["inverse_flattening"]` to `"inverse_flattening"` (was `"flattening"`).

### CODE REVIEW

**Comment by FussyDuck** [comment]
[CLA assistant check](https://bol-claassistant.ecmwf.int:443/ecmwf/earthkit-data?pullRequest=503) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://bol-claassistant.ecmwf.int:443/ecmwf/earthkit-data?pullRequest=503) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://bol-claassistant.ecmwf.int:443/check/ecmwf/earthkit-data?pullRequest=503) it.</sub>

File: src/earthkit/data/readers/geotiff.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+        self._not_implemented()
+
+    def bounding_box(self):
+        left, bottom, right, top = self._ds.rio.transform_bounds("EPSG:4326")
+        return BoundingBox(north=top, west=left, south=bottom, east=right)
+
+    def gridspec(self):
+        self._not_implemented()
+
+    def resolution(self):
```
Comment (sandorkertesz) [inline src/earthkit/data/readers/geotiff.py:77]: This is a tricky method. For GRIB, returns a single value, and asserts if the resolution is different in x and y directions.

File: src/earthkit/data/readers/geotiff.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+from functools import cached_property
+
+import numpy as np
+
+from ..core.fieldlist import Field
+from ..core.fieldlist import FieldList
+from ..core.geography import Geography
+from ..core.metadata import RawMetadata
+from ..utils.bbox import BoundingBox
+from ..utils.projections import Projection
```
Comment (sandorkertesz) [inline src/earthkit/data/readers/geotiff.py:20]: Projection should be a local import in the methods that use it. With this we can delay importing e.g. catropy until it is actually used.

**Comment by sandorkertesz** [comment]
@chpolste , thank you very much for this development.  I tested it and works nicely.  I made a few comments, once these and the branch conflict are sorted out the PR can be merged. 

As for the documentation, what notebook example do you think should be added? What should it be highlight?

File: tests/readers/test_geotiff_reader.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import pytest
+
+from earthkit.data import from_source
+from earthkit.data.readers.geotiff import GeoTIFFField
+from earthkit.data.testing import NO_RIOXARRAY
+from earthkit.data.testing import earthkit_test_data_file
+from earthkit.data.utils.projections import TransverseMercator
+
+
+@pytest.mark.skipif(NO_RIOXARRAY, "rioxarray not available")
```
Comment (sandorkertesz) [inline tests/readers/test_geotiff_reader.py:21]: I think this should be: 

```python
@pytest.mark.skipif(NO_RIOXARRAY, reason="rioxarray not available")
```

**Pull request merged**